homeassistant/components/light/xiaomi_miio.py

```diff
@@ -227,8 +272,49 @@ def _try_command(self, mask_error, func, *args, **kwargs):
             _LOGGER.error(mask_error, exc)
             return False
 
-    @asyncio.coroutine
-    def async_turn_on(self, **kwargs):
+    async def async_turn_on(self, **kwargs):
+        """Turn the light on."""
+        await self._try_command(
+            "Turning the light on failed.", self._light.on)
+
+    async def async_turn_off(self, **kwargs):
+        """Turn the light off."""
+        await self._try_command(
+            "Turning the light off failed.", self._light.off)
+
+    async def async_update(self):
+        """Fetch state from the device."""
+        from miio import DeviceException
+        try:
+            state = await self.hass.async_add_job(self._light.status)
+            _LOGGER.debug("Got new state: %s", state)
+
+            self._state = state.is_on
+
+        except DeviceException as ex:
+            self._state = None
+            _LOGGER.error("Got exception while fetching the state: %s", ex)
+
+
+class XiaomiPhilipsGenericLight(XiaomiPhilipsAbstractLight):
+    """Representation of a Xiaomi Philips Light."""
+
+    def __init__(self, name, light, model):
+        """Initialize the light device."""
+        XiaomiPhilipsAbstractLight.__init__(self, name, light, model)
+
+        self._color_temp = None
+        self._state_attrs.update({
+            ATTR_SCENE: None,
+            ATTR_DELAYED_TURN_OFF: None,
+        })
+
+    @property
+    def supported_features(self):
+        """Return the supported features."""
+        return SUPPORT_FLAGS_GENERIC
```

Comment:
Do not do this. You can't make up your own feature flags. Only use the ones as provided by the component and nothing extra.

Reply:
What do you think about the new solution?

Reply:
Better. I still think that naming it `SUPPORT_*` is easily confused with the real supported features